### ISSUE TITLE
SSH hanging with SSH connection manager

### ISSUE THREAD

**Issue by Jule-** [Eugeny/terminus#575]
- Version: 1.0.65
- Platform: win32 10.0.17134
- Plugins: none

---

When I try to use connection manager, the ssh connection is established since I see the entry log of my server but I end up with no bash prompt. Just cursor blinking at the start of a new line and nothing that I can type (even Ctrl+C is not working).

In my debug log I have these 2 errors:

```log
preload.js:607 Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:78:25)
    at writeGeneric (internal/stream_base_commons.js:73:3)
    at Socket._writeGeneric (net.js:713:5)
    at Socket._write (net.js:725:8)
    at doWrite (_stream_writable.js:410:12)
    at writeOrBuffer (_stream_writable.js:394:5)
    at Socket.Writable.write (_stream_writable.js:294:11)
    at Socket.Writable.end (_stream_writable.js:579:10)
    at Socket.end (net.js:544:31)
    at PageantSock.write (C:\Users\jule\AppData\Local\terminus\app-1.0.65\resources\builtin-plugins\terminus-ssh\dist\index.js:17381:16)
process.on @ preload.js:607
```

```log
preload.js:607 Error: spawn C:\util\pagent.exe ENOENT
    at Process.ChildProcess._handle.onexit (/C:/Users/jule/AppData/Local/terminus/app-1.0.65/resources/app.asar/dist/internal/child_process.js:232)
    at onErrorNT (/C:/Users/jule/AppData/Local/terminus/app-1.0.65/resources/app.asar/dist/internal/child_process.js:407)
    at process._tickCallback (/C:/Users/jule/AppData/Local/terminus/app-1.0.65/resources/app.asar/dist/internal/process/next_tick.js:63)
```

Indeed I do not have pagent here `C:\util\pagent.exe` and I don't know why Terminus is trying to reach it here or even why Terminus is trying to spawn it.

What is the terminal spawn behind ssh connection manager? When I use `ssh` from my WSL terminal it works like expected.

**Comment by Eugeny** [comment]
Duplicate of #328

**Comment by Jule-** [comment]
https://github.com/Eugeny/terminus/issues/328#issue-315736035:
> It is currently not possible to connect to ssh servers that has option AllowAgentForwarding no set.
>
> When connecting to such server terminus shows error: Could not connect: Error: Unable to request agent forwarding

- I do not have the error mentioned.
- I have tried setting AllowAgentForwarding to yes with no luck.


https://github.com/Eugeny/terminus/issues/328#issuecomment-447576938:
> Now fixed together with #496


https://github.com/Eugeny/terminus/issues/496#issuecomment-440791216:
> Some crazy shit happens here between ssh2 nodejs lib (who use own copy of pag{e}ant) and and great Electron Framework on which this greatest terminal app of all time build)). On certain circumstances ssh2 go not to own folder, but for root folder which is C:/, and can't find any bit own pagent.
> 
> So workaround for those who desperately need "git push" is - to put that file where it want. This file https://github.com/mscdex/ssh2/blob/master/util/pagent.exe to that place C:\util\pagent.exe
> 
> And you need Putty SSH authentication agent pageant.exe already running.

> Eugeny closed this in https://github.com/Eugeny/terminus/commit/384744ec44a9241df687bd691c42ca5b7498c3c9 3 days ago

Can you explain in what way you have fixed this issue please? For what I've understood you are just checking `pagent` running process. Do I need to have it running? Why? Even with `pagent` running I have same issue.
What prerequisites I do not fulfill?

From my point of view it is still not working and if it is fixed, it is not seamless.

Thanks for your explainations.

**Comment by Eugeny** [comment]
It seems that in a few cases (under at this moment unknown circumstances), `ssh2` will look for its Pageant connector binary in c:\util instead of where it's actually located. I've added a workaround that doesn't automatically enable Pageant integration unless Pageant is actually running.
